### PR TITLE
[WIP] - Use the same demangling functions

### PR DESCRIPTION
**Pull request by forki**
@vasily-kirichenko could you please review? thanks

### CODE REVIEW

File: src/fsharp/PrettyNaming.fs [hunk 1]
```diff
@@ -131,10 +131,20 @@ module internal Microsoft.FSharp.Compiler.PrettyNaming
         let nameLen = name.Length
         let rec loop i = (i < nameLen && (opCharSet.Contains(name.[i]) || loop (i+1)))
         loop 0
+    
+    /// Returns the plain name of a given identifier, operator or double backticked name, 
+    /// e.g. "|>>" instead of "( |>> )" or "long identifier" instead of "( long identifier )".
+    let DemangleBacktickedName (name:string) =
```
Comment (dsyme) [inline src/fsharp/PrettyNaming.fs]: The name of the function seems odd - This seems to deal with parenthesized names, not backtick names?

Comment (forki) [inline src/fsharp/PrettyNaming.fs]: how could we name it?

Comment (dsyme) [inline src/fsharp/PrettyNaming.fs]: I'd need to look more closely.  Something smells here - somehow it seems wrong that the parentheses are present at all for these names - IIRC the parentheses are only added for output purposes - but now we are removing them and adding the backticks.  Something just feels wrong, like we have multiple versions of "display text for identifier" and aren't distinguishing them very well.

In the meantime something like ``DemangleParentheseesForBacktickIdentifier`` or something I suppose.  But the difficulty in naming seems to indicate some other problem.

It's not a big deal, and I've not reviewed this code in a while, just glancing.

File: src/fsharp/PrettyNaming.fs [hunk 2]
```diff
@@ -131,10 +131,20 @@ module internal Microsoft.FSharp.Compiler.PrettyNaming
...
+        if name.StartsWith "( " && name.EndsWith " )" then
+            name.[2..name.Length - 3]
+        else name
+    
+    /// Wraps double backticks around a name if it contains a space.
+    let NameUsedInCode (name:string) = if name.Contains " " then "``" + name + "``" else name
```
Comment (saul) [inline src/fsharp/PrettyNaming.fs]: The logic for using double backticks is more complicated than this. Take a look at QuoteIdentifierIfNeeded in `src/fsharp/lexhelp.fs` - ideally use that function instead

Comment (vasily-kirichenko) [inline src/fsharp/PrettyNaming.fs]: Exactly. I've found this function today being developing https://github.com/Microsoft/visualfsharp/pull/2236

Comment (forki) [inline src/fsharp/PrettyNaming.fs]: @vasily-kirichenko @saul can one of you send a pull request?

**Comment by KevinRansom** [comment]
@dsyme , @forki -- this is labelled wip, but no movement for 22 days is it ready?

**Comment by forki** [comment]
I'm coming back to this in 2 weeks

**Comment by KevinRansom** [comment]
@forki 
Sweet .. thanks mate.

**Comment by forki** [comment]
this is deverged too much from master

**Pull request closed without merging**